src/app/court/court.component.ts

```diff
@@ -1,14 +1,185 @@
+  showCourtForm: boolean;
+  showCourtSelector: boolean;
+  showPageMessage: boolean;
+  courtCtrl: FormControl = new FormControl();
+  filteredCourts: Observable<any[]>;
+  court: Court;
+  courts: Court[];
+  judges: Judge[];
+  pageMessage: String = '';
+  currentAction: PageAction = PageAction.NONE;
+  courtSelectorPlaceholder: String = '';
+
+  addCourt() {
+    this.court = new Court();
+    this.court.citationExpiresAfterDays = -1;
+    this.citationDoesNotExpire = true;
+    this.currentAction = PageAction.ADD;
+    this.showPageMessage = false;
+    this.showCourtForm = true;
+    this.showCourtSelector = false;
+  }
+
+  editCourt() {
+    this.currentAction = PageAction.EDIT;
+    this.assignFilterCourts();
+    this.courtSelectorPlaceholder = 'Select court to be edited';
+    this.showPageMessage = false;
+    this.showCourtForm = false;
+    this.showCourtSelector = true;
+    this.courtCtrl.setValue('');
+  }
+
+  deleteCourt() {
+    this.currentAction = PageAction.DELETE;
+    this.assignFilterCourts();
+    this.courtSelectorPlaceholder = 'Select court to be deleted';
+    this.showPageMessage = false;
+    this.showCourtForm = false;
+    this.showCourtSelector = true;
+    this.courtCtrl.setValue('');
+  }
+
+  saveForm() {
+    if (this.citationDoesNotExpire === true) {
+      this.court.citationExpiresAfterDays = -1;
+    }else {
+      this.court.citationExpiresAfterDays = this.daysUntilCitationExpires;
+    }
+    this.courtService.save(this.court).subscribe((court) => {
+      this.pageMessage = court.name + ' saved';
+      this.showPageMessage = true;
+      this.showCourtForm = false;
+      this.showCourtSelector = false;
+      this.currentAction = PageAction.NONE;
+
+      if (this.court.id == null) {
+        this.courts.push(court);
+      }
+    });
+  }
+
+  formatPhoneNumber() {
+    // formats a 10 digit phone number to use just '.' as the separation character
+    if (this.court.phone !== '') {
+      this.court.phone = this.court.phone.replace(/[- .()]/g, '');
+      this.court.phone = this.court.phone.slice(0, 6) + '.' + this.court.phone.slice(6);
+      this.court.phone = this.court.phone.slice(0, 3) + '.' + this.court.phone.slice(3);
+    }
+  }
+
+  private assignCourts(): void {
+    this.courtService.getAll().subscribe(
+      (courts) => {
+        this.courts = courts;
+      }
+    );
+  }
+
+  onCourtSelected(event: MatAutocompleteSelectedEvent) {
+    this.court = _.find(this.courts, {'name': event.option.value});
+
+    switch ( this.currentAction ) {
+      case PageAction.DELETE:
+        const index = this.courts.indexOf(this.court);
+        const dialogRef = this.dialog.open(ConfirmDialogComponent, {
+          width: '250px',
+          data: {message: 'Remove \'' + this.court.name + '\'? This will remove the court from the database permanently.' }
+        });
 
-  constructor() { }
+        dialogRef.afterClosed().subscribe(result => {
+          if (result === true) {
```

Comment:
Not a biggie but a simple 'truthy' check should suffice here: `if(result) {...}`